File: backend/Domain/Reservation.cs
namespace Backend.Domain;

public class Reservation
{
    public int Id { get; set; }
    public int RoomId { get; set; }
    public Room? Room { get; set; }
    public string GuestName { get; set; } = "";
    public string GuestEmail { get; set; } = "";
    public DateOnly CheckIn { get; set; }
    public DateOnly CheckOut { get; set; }
    public DateTime CreatedAt { get; set; }
    public int? UserId { get; set; }
    public User? User { get; set; }
}

File: frontend/src/app/models/reservation.ts
export interface Reservation {
  id: number; roomId: number; roomNumber: string;
  guestName: string; guestEmail: string; checkIn: string; checkOut: string; createdAt: string;
}
export interface CreateReservation {
  roomId: number; checkIn: string; checkOut: string;
}
